doc for callback.h and callbackSetQueueSize sanity check

mdavidsaver opened the pull request:
Add doc. comments to `callback.h`.  The only functional change is an added sanity check for `callbackSetQueueSize()` to reject non-positive sizes.

~~I'm making this a draft PR pending an answer to #718.~~

mdavidsaver committed:
doc and callbackSetQueueSize sanity check

anjohnson commented:
Most of the newer Doxygen annotations in `modules/database/src/ioc/db` use `@` instead of `\` to introduce keywords; I think in one Codeathon we suggested that because `@` is easier to grep for, although both work so this isn't a blocker.

This file isn't appearing in the RTD documentation yet. The header needs an `@file ` annotation adding near the top for Doxygen to process it, and then Sphinx needs `documentation/Makefile.headers` to add this to the "IOC Database C/C++ APIs" index:
```Makefile
database_HEADERS += callback
```
I want to automate the latter (it should be reasonably straightforward), but we just haven't got around to doing it yet.

anjohnson commented:
Core: Approved after modification to `documentation/Makefile.headers` as above.

mdavidsaver commented:
Merged as of c2140ad0f32cd510d8a41809213a016c9b53ceea

I changed the doxygen tag style from `\` to `@` (which qtcreator wants to do anyway), and added some additional discussion of timer allocation to `callbackRequestDelayed()`.

I tested a local `make -C documentation doxygen` and see that a page for `callback.h` is created.
